### PR TITLE
separate the upgrade module from apply module.

### PR DESCRIPTION
**Pull request by bxy4543**
separate the upgrade module from apply module.

### CODE REVIEW

File: apply/applydriver/local.go [hunk 1]
```diff
@@ -213,11 +231,55 @@ func (c *Applier) upgradeCluster(mj, nj []string) error {
 		return err
 	}
 
-	logger.Info("Succeeded in upgrading current cluster from version(%s) to version(%s)", info.GitVersion, clusterMetadata.Version)
+	logger.Info("Succeeded in upgrading current cluster from version(%s) to version(%s)", c.CurrentClusterInfo.GitVersion, upgradeImgMeta.Version)
+	return utils.SaveClusterInfoToFile(c.ClusterDesired, c.ClusterDesired.Name)
+}
+
+func (c *Applier) CheckDiffImage(imageName string) error {
+	if c.ClusterDesired.Spec.Image == imageName {
```
Comment (fanux) [inline apply/applydriver/local.go:239]: Check current cluster version == ClusterImage kubernetes version

Comment (bxy4543) [reply apply/applydriver/local.go:239]: > Check current cluster version == ClusterImage kubernetes version

This is already implemented in line 219.

**Comment by fanux** [comment]
What if user want to use Clusterfile to upgrade the cluster?

1. edit Clusterfile kubernetes:v1.19.8->kubernetes:v1.20.0
2. kubectl apply -f Clusterfile

**Comment by bxy4543** [comment]
> What if user want to use Clusterfile to upgrade the cluster?
> 
> 1. edit Clusterfile kubernetes:v1.19.8->kubernetes:v1.20.0
> 2. kubectl apply -f Clusterfile

done

**Review by fanux — APPROVED** [review]

**Pull request merged**